### PR TITLE
intro section first draft

### PR DESCRIPTION
**Pull request by pat-rogers**

### CODE REVIEW

File: content/courses/intro-to-embedded-sys-prog/chapters/introduction.rst [hunk 1]
```diff
@@ -8,6 +8,183 @@ Introduction
...
+We say "within reason" because portability is a difficult goal for 
+embedded systems. That's because the hardware is so much a part of the 
+application itself, rather than being abstracted away as in a 
+general-purpose application. That said, the hardware details can be 
+managed in Ada so that portability is maximized to the extent possible 
+for the application. 
+
+Strictly speaking, not all software can or should be absolutely 
+portable! If a specific device is required, well, the program won't work 
+with some other device. But tot he extent possible portability is
```
Comment (gusthoff) [inline content/courses/intro-to-embedded-sys-prog/chapters/introduction.rst:121]:
```suggestion
with some other device. But to the extent possible portability is 
```

File: content/courses/intro-to-embedded-sys-prog/chapters/introduction.rst [hunk 2]
```diff
@@ -8,6 +8,183 @@ Introduction
...
+This language design principle is known as the "separation principle." 
+What's being separated? The low-level, less portable aspects of some 
+piece of code are separated from the usage of that piece of code. 
+
+Don't confuse this with hiding unnecessary implementation details via 
+compile-time visibility control (i.e., information hiding and 
+encapsulation). That certainly should be done too. Instead, because of 
+the separation principle, we specify the low-level properties of 
+something once, when we declare it. From then on, we can use regular Ada 
+code to interact with it. That way the bulk of the code -- the usage --
```
Comment (gusthoff) [inline content/courses/intro-to-embedded-sys-prog/chapters/introduction.rst:141]:
```suggestion
code to interact with it. That way the bulk of the code |mdash| the usage |mdash| 
```

File: content/courses/intro-to-embedded-sys-prog/chapters/introduction.rst (hunk 1 above)
Comment (pat-rogers) [reply content/courses/intro-to-embedded-sys-prog/chapters/introduction.rst:121]: Got it.

File: content/courses/intro-to-embedded-sys-prog/chapters/introduction.rst (hunk 2 above)
Comment (pat-rogers) [reply content/courses/intro-to-embedded-sys-prog/chapters/introduction.rst:141]: Got it.

**Pull request closed without merging**